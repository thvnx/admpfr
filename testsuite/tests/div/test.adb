with Ada.Text_IO; use Ada.Text_IO;
with Admpfr;      use Admpfr;

procedure Test is
   M, N, O : Mpfloat;
begin
   N.Set (Float (1.0));
   O.Set (Float (4.0));

   M.Div (N, O);
   Put_Line (M'Image);

   M.Div (N, 5);
   Put_Line (M'Image);

   M.Div (N, 6.0);
   Put_Line (M'Image);

   M.Div (5, N);
   Put_Line (M'Image);

   M.Div (6.0, N);
   Put_Line (M'Image);

   M.Div_2 (N, 8);
   Put_Line (M'Image);
end Test;
