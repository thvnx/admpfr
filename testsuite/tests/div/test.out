2.5000000000000000e-01
2.0000000000000001e-01
1.6666666666666666e-01
5.0000000000000000e+00
6.0000000000000000e+00
